Code/Framework/AzCore/AzCore/Task/TaskExecutor.h

```diff
@@ -53,6 +54,7 @@ namespace AZ
             // The pointer to the parent graph is set only if it is retained
             TaskGraph* m_parent = nullptr;
             AZStd::atomic<uint32_t> m_remaining;
+            const char* m_parent_label;
```

Comment:
Same here - "m_parentLabel".

Reply:
It'd be safer for this case to use `AZStd::string_view` instead (both in member and the argument passed). That will make it clear that this class doesn't own the string and that the caller has to make sure the string passed needs to live as long as this class lives.

If that's not the case, then this class needs to hold a copy of the label using `AZStd::string`.

Making it explicit using a string view can avoid errors when constructing this class by passing a string that doesn't outlive the instance.

Same for all the other places.

Reply:
String views don't imply null termination though and we may want to log or do other things with these labels that interact with a C interface. I actually very much dislike string views in most places because of this irritating pitfall.

Reply:
Personally I'd be inclined to just KISS, add a comment warning that the string passed in must outlive the lifetime of the task since every single example you have uses a static string and is safe. The very first thing people will ever do will be to copy an existing example of usage. Doing deep string copies seems overkill, and similarly string_view is dangerous and with high frequency leads to unsafe or broken code.

Reply:
Thanks that's honestly my preferred approach also, I can add a comment. Generally, things that crash immediately I'm not as inclined to go out of my way to provide all the template guardrails or safety measures on, especially on a hot path.